plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ConnectionMetadataUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public final class ConnectionMetadataUtils
+{
+    private static Boolean UPPERCASE;
```

Comment:
Even if correct, due to classloader isolation, caching on static field in util class looks like a major code smell.
Can't we just use a field in `BaseJdbcClient`?